### DISCUSSION TITLE
Using FastAPI w/o gunicorn/WSGI?

### DISCUSSION THREAD

**Discussion by ichorid** [fastapi/fastapi#2524] [Questions] [answered]
Hi!

Is it possible to use FastAPI without gunicorn or another ASGI server, e.g. on top of raw AIOHTTP?
I have to be able to correctly parse URIs containing URLencoded `/` symbols, and there seems to be [an ASGI specification problem](https://github.com/encode/starlette/issues/826) that's not going to be fixed anytime soon. Also, my app must be able to run on every platform including Windows, and Uvloop is not available there.

What will be the best strategy to tackle this?

**Comment by Mause** [comment]
aiohttp is a library, so I'm going to assume you meant raw asgi.

Ultimately there is no such thing as "raw asgi", you still need a server that runs your application and actually handles the http layer and binding to sockets, etc.

**Comment by ichorid** [comment, marked as answer]
Found this bare-bone ASGI server working on top of AIOHTTP. It does just what I need...
https://github.com/mosquito/aiohttp-asgi
